services/management/service.go

```diff
@@ -92,15 +92,42 @@ func (s *ServiceService) RemoveService(ctx context.Context, req *managementpb.Re
 				return err
 			}
 			if agent.PMMAgentID != nil {
-				pmmAgentIDs[pointer.GetString(agent.PMMAgentID)] = true
+				pmmAgentIDs[pointer.GetString(agent.PMMAgentID)] = struct{}{}
 			} else {
 				reloadPrometheusConfig = true
 			}
 		}
+
 		err = models.RemoveService(tx.Querier, service.ServiceID, models.RemoveCascade)
 		if err != nil {
 			return err
 		}
+
+		node, err := models.FindNodeByID(s.db.Querier, service.NodeID)
+		if err != nil {
+			return err
+		}
+
+		// For RDS and Azure remove also node.
+		if node.NodeType == models.RemoteRDSNodeType || node.NodeType == models.RemoteAzureDatabaseNodeType {
+			agents, err = models.FindAgents(tx.Querier, models.AgentFilters{NodeID: node.NodeID})
+			if err != nil {
+				return err
+			}
+			for _, a := range agents {
+				_, err := models.RemoveAgent(s.db.Querier, a.AgentID, models.RemoveRestrict)
+				if err != nil {
+					return err
+				}
+				if a.PMMAgentID != nil {
+					pmmAgentIDs[pointer.GetString(a.PMMAgentID)] = struct{}{}
+				}
+			}
+			if err = models.RemoveNode(tx.Querier, node.NodeID, models.RemoveCascade); err != nil {
+				return err
+			}
+		}
```

Comment:
looks like we can move this into `RemoveService` and check for `models.RemoveCascade`

Reply:
There is a problem with that because we collecting `pmmAgentIDs` in this loop. So if we will move all this stuff inside `RemoveService` we should return a list of pmm agents that require state update.